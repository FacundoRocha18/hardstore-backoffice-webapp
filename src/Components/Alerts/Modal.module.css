.container {
	background-color: rgba(0, 0, 0, 0.7);
	display: flex;
	align-items: center;
	justify-content: center;
	position: fixed;
	bottom: 0;
	left: 0;
	width: 100%;
	overflow: hidden;
	z-index: 1000;
	transition: height .3s ease;
}

.container.inactive {
	height: 0px;
	transition: height .3s ease;
}

.container.active {
	height: 100%;
	transition: height .3s ease;
}

.modal {
	background-color: rgb(245, 245, 245);
	border-radius: 5px;
	height: fit-content;
	width: fit-content;
	padding: 20px;
	z-index: 2000;
}

.body {
	display: flex;
	flex-direction: column;
	gap: 20px;
}

.body h4 {
	color: var(--text-dark);
}

.buttonContainer {
	display: grid;
	grid-template-columns: repeat(2, 1fr);
	gap: 20px;
}

.buttonContainer button {
	width: 100%;
}
