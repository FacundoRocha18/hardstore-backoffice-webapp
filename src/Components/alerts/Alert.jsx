import React from 'react'

/* CSS Styles --------------------- */
import style from './alert.module.css'
import css from 'classnames'

/* Components --------------------- */
import { Button } from '../inputs/Button'

export const Alert = ({ type, title, message, isActive, close }) => {

	const icon = selectIcon(type);

	return (
		<>
			<section className={css(style.container, isActive ? style.active : style.inactive)}>
				<div className={css(style.head, style[type])}>
					<span className="material-icons-outlined">
						{
							icon
						}
					</span>
				</div>
				<div className={css(style.body, isActive ? style.active : style.inactive)}>
					<div className={css(style.message)}>
						<h4>
							{title}
						</h4>
						<p>{message}</p>
					</div>
					<div className={css(style.btn_container)}>
						<Button variant={'close-btn'} func={close} >
							<span className="material-icons-outlined">
								close
							</span>
						</Button>
					</div>
				</div>
			</section>
		</>
	)
}

const selectIcon = (alertType) => {

	const icons = [
		{
			type: 'info',
			icon: 'info'
		},
		{
			type: 'error',
			icon: 'error_outline'
		},
		{
			type: 'warning',
			icon: 'warning_amber'
		},
		{
			type: 'success',
			icon: 'check_circle_outline'
		}
	]

	let alertIcon;

	for (let i = 0; i < icons.length; i++) {

		const { type, icon } = icons[i]

		if (type === alertType) {
			alertIcon = icon
		}
	}

	return alertIcon;
}
