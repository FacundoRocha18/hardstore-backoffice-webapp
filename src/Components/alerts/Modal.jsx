import React, { useState } from 'react'

/* Components --------------------- */
import { Button, Loading } from '..'

/* CSS Styles --------------------- */
import style from './Modal.module.css'
import css from 'classnames'

export const Modal = ({ isActive, setIsActive, onLogout, newAlert }) => {

	const [showLoading, setLoading] = useState();

	const handleLogout = () => {

		newAlert('Se ha cerrado correctamente la sesion', 'Se lo redigira a la pagina de inicio de sesión en unos segundos', 'info')

		setLoading(true)

		setTimeout(() => {
			return onLogout()
		}, 2500)
	}

	if (showLoading) {
		return (
			<div className={css(style.container, isActive ? style.active : style.inactive)} >
				<Loading />
			</div >
		)
	}

	return (
		<div className={css(style.container, isActive ? style.active : style.inactive)}  >
			<div className={css(style.modal)}>
				<div className={style.body}>
					<h4>¿Estás seguro que quieres cerrar la sesión?</h4>
					<div className={style.buttonContainer}>
						<Button variant='accept' show={true} func={handleLogout} >
							<p>
								Si, cerrar sesión
							</p>
						</Button>
						<Button variant='decline' show={true} func={() => setIsActive(false)} >
							<p>
								Cancelar
							</p>
						</Button>
					</div>
				</div>
			</div>
		</div>
	)
}
