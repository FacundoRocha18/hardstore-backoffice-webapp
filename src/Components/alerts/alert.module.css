.container {
	align-items: center;
	background-color: rgb(245, 245, 245);
	border-radius: 5px;
	bottom: 30px;
	display: flex;
	position: fixed;
	right: 30px;
	overflow: hidden;
	z-index: 1000;
}

.container.active {
	overflow: hidden;
	width: 500px;
	height: 100px;
	transition: all .5s ease-in-out ;
}

.container.inactive {
	width: 0px;
	height: 0px;
	overflow: hidden;
	transition: all .5s ease-in-out ;
}

.icon {
	align-items: center;
	display: flex;
	height: 100%;
	justify-content: center;
	min-width: 75px;
}

.icon span {
	color: var(--text-main-color);
	color: var(--text-dark);
	font-size: 35px;
}

.container.inactive .icon span {
	font-size: 0px;
	transition: all .5s ease-in-out ;
}

.icon.info {
	background-color: var(--blue);
}

.icon.success {
	background-color: var(--green);
}

.icon.success span {
	color: var(--text-dark);
}

.icon.warning {
	background-color: var(--yellow);
}

.icon.warning span {
	color: var(--text-dark);
}

.icon.error {
	background-color: var(--red);
}

.body {
	align-items: center;
	display: flex;
	gap: 20px;
	justify-content: space-between;
}

.body.inactive {
	width: 0px;
	overflow: hidden;
	padding: 0px;
}

.body.active {
	overflow: hidden;
	width: 100%;
	padding: 20px;
}

.message {
	width: 100%;
}

.message,
.message h4,
.message p {
	color: var(--text-dark);
	text-align: left;
	white-space: nowrap;
	width: fit-content;
	padding: 0;
}

.message.inactive {
	width: 0px;
	overflow: hidden;
	padding: 0px;
}

.message.active {
	width: fit-content;
	overflow: hidden;
}

.btn_container {
	align-items: center;
	display: flex;
	justify-content: center;
	min-width: 50px;
	min-height: 50px;
}