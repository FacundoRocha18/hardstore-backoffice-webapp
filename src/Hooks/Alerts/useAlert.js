import { useState, useEffect } from "react";

export const useAlert = () => {

	const [ isActive, setIsActive ] = useState(false);

  const [title, setTitle] = useState('Some title');
	
  const [message, setMessage] = useState('Some message');

  const [type, setType] = useState(null);

  useEffect(() => {

    const timer = setTimeout(() => close(), 3000);

    return () => {
      clearTimeout(timer);
    }
  })

  const show = (title, message, type) => {

		setTitle(title);
    setMessage(message);
    setType(type);
    setIsActive(true);
		console.log(isActive)	
  };

  const close = () => setIsActive(false);

  return ({
    isActive: isActive,
		title: title,
    message: message,
    type: type,
    newAlert: show,
    closeAlert: close
  })
}
