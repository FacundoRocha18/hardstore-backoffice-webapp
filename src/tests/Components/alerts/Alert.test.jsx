import { Alert } from '../../../Components/alerts/Alert';
import { fireEvent, render, screen } from '@testing-library/react';
import { useAlert } from '../../../Hooks/Alerts/useAlert'

describe('Tests on <Alert /> component', () => {

	test('should match the snapshot', () => {

		const { container } = render(
			<Alert
				type='info'
				title='Inicio de sesion exitoso'
				message='Bienvenido Facundo.'
			/>
		)

		expect(container).toMatchSnapshot();
	})

	test('should show a title, message and icon (based on alert variant)', () => {
		render(
			<Alert
				type='success'
				title='Inicio de sesion exitoso'
				message='Bienvenido Facundo.'
				isActive={true}
			/>
		)

		const icon = screen.getAllByTestId('test-icon');
		const title = screen.getByTestId('test-title');
		const message = screen.getByTestId('test-message')

		console.log(typeof icon[0])

		expect(icon[0].innerHTML).toEqual(expect.any(String))
		expect(message.innerHTML).toEqual(expect.any(String))
		expect(title.innerHTML).toEqual(expect.any(String))
	})

	/* test('should close the alert', () => {

		render(
			<Alert
				type='success'
				title='Inicio de sesion exitoso'
				message='Bienvenido Facundo'
				isActive={true}
				close={close}
			/>
		)
		fireEvent.click(screen.getByTestId('test-close-button'))
		expect(isActive).toEqual(false)
	}) */
})