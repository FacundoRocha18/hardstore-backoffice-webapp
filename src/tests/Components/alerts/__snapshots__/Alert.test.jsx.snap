// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Alert /> component should match the snapshot 1`] = `
<div>
  <section
    class="container inactive"
  >
    <div
      class="head info"
    >
      <span
        class="material-icons-outlined"
        data-testid="test-icon"
      >
        info
      </span>
    </div>
    <div
      class="body inactive"
    >
      <div
        class="message"
      >
        <h4
          data-testid="test-title"
        >
          Inicio de sesion exitoso
        </h4>
        <p
          data-testid="test-message"
        >
          Bienvenido Facundo.
        </p>
      </div>
      <div
        class="btn_container"
      >
        <button
          class="button close-btn hidden"
          data-testid="test-icon"
        >
          <span
            class="material-icons-outlined"
          >
            close
          </span>
        </button>
      </div>
    </div>
  </section>
</div>
`;
